### PR TITLE
Stats Widget: Add 7 day highlights

### PR DESCRIPTION
**Pull request by kangzj**
Related to #75103 
## Proposed Changes

* Add the `7 Day Highlights` section.
* Apply `top-posts` and `referrers` API queries for data.

截圖 2023-04-08 上午4 54 29

## Testing Instructions

* Open your local Jetpack site.
* Check out the branch of the Jetpack to the latest `trunk`.
* Build Jetpack if necessary: `jetpack build plugins/jetpack`.
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` under `calypso/apps/odyssey-stats`.
* Navigate to `/wp-admin/index.php?odyssey_widget=1`.
* Ensure the `7 Day Highlights` section of the Stats Widget displays in line with the design.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by kangzj** [comment]
The PR needs a merge or rebase to get rid of the base code from https://github.com/Automattic/wp-calypso/pull/75190

File: apps/odyssey-stats/src/widget/highlights.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { useTranslate } from 'i18n-calypso';
+import moment from 'moment';
+import useReferrersQuery from '../hooks/use-referrers-query';
+import useTopPostsQuery from '../hooks/use-top-posts-query';
+
+import './hightlights.scss';
+
+function TopColumn( { items, viewAllUrl, viewAllText, title, className = null } ) {
+	const translate = useTranslate();
+	// TODO: add a placeholder state
```
Comment (jsnmoon) [inline apps/odyssey-stats/src/widget/highlights.jsx:11]: What would this placeholder state do?

File: apps/odyssey-stats/src/widget/highlights.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import moment from 'moment';
+import useReferrersQuery from '../hooks/use-referrers-query';
+import useTopPostsQuery from '../hooks/use-top-posts-query';
+
+import './hightlights.scss';
+
+function TopColumn( { items, viewAllUrl, viewAllText, title, className = null } ) {
```
Comment (jsnmoon) [inline apps/odyssey-stats/src/widget/highlights.jsx:9]: For a follow-up PR: This component would benefit from loading state handling. Currently, it says "Sorry, nothing to report"; we can have it say something like "Loading..." instead.

File: apps/odyssey-stats/src/widget/highlights.jsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		<div className="stats-widget-highlights">
+			<div className="stats-widget-highlights__header">
+				<label>{ translate( '7 Day Highlights' ) }</label>
+				<a href={ odysseyStatsBaseUrl }>{ translate( 'View detailed stats' ) }</a>
+			</div>
+			<div className="stats-widget-highlights__body">
+				<TopColumn
+					className="stats-widget-highlights__column"
+					title={ translate( 'Top Posts & Pages' ) }
+					viewAllUrl={ odysseyStatsBaseUrl }
```
Comment (jsnmoon) [inline apps/odyssey-stats/src/widget/highlights.jsx:57]: For a follow-up PR: Having multiple links to the same URL (`odysseyStatsBaseUrl`) is redundant. Perhaps these should link deeply into the relevant summary pages, like so:
- `/wp-admin/admin.php?page=stats#!/stats/day/posts/:siteSlug`
- `/wp-admin/admin.php?page=stats#!/stats/day/referrers/:siteSlug`

**Review by jsnmoon — APPROVED** [review]
This works exactly as described. Nice work!

File: apps/odyssey-stats/src/widget/highlights.jsx (hunk 1 above)
Comment (dognose24) [reply apps/odyssey-stats/src/widget/highlights.jsx:11]: It is like the loading spinner on the chart, the same as [your mentioned comment](https://github.com/Automattic/wp-calypso/pull/75323#discussion_r1162138268). 😄

File: apps/odyssey-stats/src/widget/highlights.jsx (hunk 3 above)
Comment (dognose24) [reply apps/odyssey-stats/src/widget/highlights.jsx:57]: Yes, respective links are more convenient for users to find those details.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7919751

Thank you @kangzj for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7919751).

**Pull request merged**